lib/epochs.js

```diff
@@ -0,0 +1,118 @@
+const { promisify: p } = require('util')
+const { fromMessageSigil, fromFeedSigil } = require('ssb-uri2')
+
+const Reduce = require('@tangle/reduce')
+const Strategy = require('@tangle/strategy')
+const OverwriteFields = require('@tangle/overwrite-fields')
+
+const getTangleUpdates = require('./tangles/get-tangle-updates')
+
+const msgPattern = [
+  '^',
+  'ssb:message',
+  '/',
+  '[a-zA-Z0-9-]+', // message "format"
+  '/',
+  '[a-zA-Z0-9-_]{43}=', // message "key"
+  '$'
+].join('')
+
+const feedPattern = [
+  '^',
+  'ssb:feed',
+  '/',
+  '[a-zA-Z0-9-_]+', // feed "format"
+  '/',
+  '[a-zA-Z0-9-_]{43}=', // feed "key"
+  '$'
+].join('')
+
+const strategy = OverwriteFields({
+  keyPattern: msgPattern,
+  valueSchema: {
+    type: 'object',
+    properties: ['author', 'epochKey'],
+    required: ['author', 'epochKey'],
+    author: {
+      type: 'string',
+      pattern: feedPattern
+    },
+    epochKey: {
+      type: 'string',
+      pattern: '[a-zA-Z0-9/\\+]{43}=' // 32 bytes in base64
```

Comment:
this and the above regex feel a bit like magic numbers, that we've written in several places. is there a way to import at least some of this from somewhere else? for the author ssb-uri2 might help?

Reply:
is it simpler maybe to do like you've suggested before and run the normal validate functions directly  on the messages around line 87?

Reply:
uhhh .... this builds the validate functions are runs them for you in `reduce`
But yes I'll try and find a more canonical source for these pattern strings.

Reply:
`ssb-uri2` does not expose its patterns / regexp. Annoying, and outside of scope of this. I'll leave a note in the code

Reply:
ok I built a module. This is your fault
https://github.com/ssbc/butt64

Reply:
wow cool!